torch/csrc/distributed/rpc/utils.cpp

```diff
@@ -606,6 +651,20 @@ Message tensorpipeDeserialize(
     tensors.emplace_back(std::move(t));
   }
 
+  if (!buffers.deviceIndices.empty()) {
+    TORCH_INTERNAL_ASSERT(
+        buffers.deviceIndices.size() == tensors.size(),
+        "Number of device indices must match the number of tensors in the "
+        "RPC message. But got ",
+        tensors.size(),
+        " tensors with ",
+        buffers.deviceIndices.size(),
+        " device indices.");
+    for (size_t i = {}; i < tensors.size(); ++i) {
+      tensors[i] = tensors[i].to(indexToDevice(buffers.deviceIndices[i]));
```

Comment:
> Calling .to(CPU) for a tensor that is already on CPU is a no-op, right?

Yep, I believe so.

> Do we maybe want to special-case this to avoid this possibly becoming a trap?

Sure, let me add a branch.